gtsam/discrete/DecisionTree-inl.h

```diff
@@ -678,7 +680,14 @@ namespace gtsam {
   }
 
   /****************************************************************************/
-  // Functor performing depth-first visit without Assignment<L> argument.
+  /**
+   * Functor performing depth-first visit without Assignment<L> argument.
+   *
+   * NOTE: We differentiate between leaves and assignments. Concretely, a 3
```

Comment:
So, what does this Functor do? Leaves, right? And should it not now pass the number of assignments captured by the leaf?

Reply:
The `VisitLeaf` functor passes the `Leaf` object which has the number of assignments recorded in it.